enterprise/frontend/src/metabase-enterprise/rich_text_editing/tiptap/extensions/CardEmbed/CardEmbedNode.tsx

```diff
@@ -62,186 +59,15 @@ import {
 import { createIdAttribute, createProseMirrorPlugin } from "../NodeIds";
 import CS from "../extensions.module.css";
 import { NativeQueryModal } from "../shared/NativeQueryModal";
+import { useDndHelpers } from "../shared/dnd/use-dnd-helpers";
 
+import { CardEmbedMenuDropdown } from "./CardEmbedMenuDropdown";
 import styles from "./CardEmbedNode.module.css";
 import { PublicDocumentCardMenu } from "./PublicDocumentCardMenu";
 import { ModifyQuestionModal } from "./modals/ModifyQuestionModal";
-import { useDndHelpers } from "./use-dnd-helpers";
 import { useUpdateCardOperations } from "./use-update-card-operations";
 import { getEmbedIndex } from "./utils";
 
-interface CardEmbedMenuContext {
-  canWrite: boolean;
-  dataset: Dataset | undefined;
-  question: Question | undefined;
-  isNativeQuestion: boolean | undefined;
-  commentsPath: string;
-  hasUnsavedChanges: boolean;
-  unresolvedCommentsCount: number;
-}
-
-interface CardEmbedMenuActions {
-  handleDownload: (opts: {
-    type: string;
-    enableFormatting: boolean;
-    enablePivot: boolean;
-  }) => void;
-  handleEditVisualizationSettings: () => void;
-  setIsModifyModalOpen: (open: boolean) => void;
-  handleReplaceQuestion: () => void;
-  handleRemoveNode: () => void;
-  handleAddSupportingText?: () => void;
-}
-
-interface CardEmbedMenuState {
-  menuView: string | null;
-  setMenuView: (view: string | null) => void;
-  isDownloadingData: boolean;
-}
-
-type CardEmbedMenuDropdownProps = CardEmbedMenuContext &
-  CardEmbedMenuActions &
-  CardEmbedMenuState;
-
-const CardEmbedMenuDropdown = ({
```

Comment:
This is moved to `enterprise/frontend/src/metabase-enterprise/rich_text_editing/tiptap/extensions/CardEmbed/CardEmbedMenuDropdown.tsx`